test/test_metadata.py

```diff
@@ -187,3 +187,146 @@ def test_compare_deleted(self):
+        del m['x']
+        self.assertEqual(len(m), 3)
+        del m['c']
+        self.assertEqual(len(m), 2)
+        #TODO: test with cover art images
+
+    def _check_mapping_update(self, m):
+        self.assertEqual(m['a'], 'b')
+        self.assertEqual(m['c'], '3')
+        self.assertEqual(m.getraw('d'), ['u', 'w'])
+        self.assertEqual(m['x'], '')
+        self.assertIn('x', m.deleted_tags)
+
+    def test_metadata_mapping_update(self):
+        # update from Metadata
+        m = self.metadata_d2
+        m2 = self.metadata_d3
+
+        del m2['x']
+        m.update(m2)
+        self._check_mapping_update(m)
+
+    def test_metadata_mapping_update_dict(self):
+        # update from dict
+        m = self.metadata_d2
+
+        d2 = {'c': 3, 'd': ['u', 'w'], 'x': ''}
+
+        m.update(d2)
+        self._check_mapping_update(m)
+
+    def test_metadata_mapping_update_tuple(self):
+        # update from tuple
+        m = self.metadata_d2
+
+        d2 = (('c', 3), ('d', ['u', 'w']), ('x', ''))
+
+        m.update(d2)
+        self._check_mapping_update(m)
+
+    def test_metadata_mapping_update_dictlike(self):
+        # update from kwargs
+        m = self.metadata_d2
+
+        m.update(c=3, d=['u', 'w'], x='')
+        self._check_mapping_update(m)
+
+    def test_metadata_mapping_update_noparam(self):
+        # update without parameter
+        m = self.metadata_d2
+
+        m.update()
+        self.assertEqual(m['a'], 'b')
+
+    def test_metadata_mapping_update_intparam(self):
+        # update without parameter
+        m = self.metadata_d2
+
+        self.assertRaises(TypeError, m.update, 123)
+
+    def test_metadata_mapping_update_strparam(self):
+        # update without parameter
+        m = self.metadata_d2
+
+        self.assertRaises(ValueError, m.update, 'abc')
+
+    def test_metadata_mapping_update_kw(self):
+        m = Metadata(tag1='a', tag2='b')
+        m.update(tag1='c')
+        self.assertEqual(m['tag1'], 'c')
+        self.assertEqual(m['tag2'], 'b')
+        m.update(tag2='')
+        self.assertIn('tag2', m.deleted_tags)
+
+    def test_metadata_mapping_update_kw_del(self):
+        m = Metadata(tag1='a', tag2='b')
+        del m['tag1']
+
+        m2 = Metadata(tag1='c', tag2='d')
+        del m2['tag2']
+
+        m.update(m2)
+        self.assertEqual(m['tag1'], 'c')
+        self.assertNotIn('tag2', m)
+        self.assertNotIn('tag1', m.deleted_tags)
+        self.assertIn('tag2', m.deleted_tags)
+
+    def test_metadata_mapping_images(self):
+        image1 = create_image(b'A', comment='A')
+        image2 = create_image(b'B', comment='B')
+
+        m1 = Metadata(a='b', length=1234, images=[image1])
+        self.assertEqual(m1.images[0], image1)
+
+        m1.append_image(image2)
+        self.assertEqual(m1.images[1], image2)
+
+        m1.remove_image(0)
+        self.assertEqual(m1.images[0], image2)
+
+        m2 = Metadata(a='c', length=4567, images=[image1])
+        m1.update(m2)
```

Comment:
In this case, current (and previous code) removes `image2` since it isn't in `m2` and `Metadata.update()` just copies the list of images of `m2` to `m1`.